cpu/producer_consumer.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Author: Abhishek Goel<huntbag@linux.vnet.ibm.com>
+#
+# Based on code by Gautham Shenoy <ego@linux.vnet.ibm.com>
+
+import os
+
+from avocado import Test
+from avocado.utils import process
+from avocado.utils import build, git
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Producer_Consumer(Test):
+
+    '''
+    Producer-consumer is a cache affinity scheduler wakeup benchmark.
+
+    :avocado: tags=cpu
+    '''
+
+    def setUp(self):
+        '''
+        Build schbench
+        Source:
+        https://github.com/gautshen/misc.git
+        '''
+        sm = SoftwareManager()
+        for package in ['gcc', 'make']:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.cancel("%s is needed for the test to be run" % package)
+        url = 'https://github.com/gautshen/misc.git'
+        git.get_repo(url, destination_dir=self.workdir)
+        self.sourcedir = os.path.join(self.workdir, 'producer_consumer')
+        os.chdir(self.sourcedir)
+        build.make(self.sourcedir)
+
+    def test(self):
+
+        perfstat = self.params.get('perfstat', default='')
+        if perfstat:
+            perfstat = 'perf stat ' + perfstat
+        pcpu = self.params.get('pcpu', default='0')
+        ccpu = self.params.get('ccpu', default='1')
+        random_seed = self.params.get('random_seed', default=6407741)
+        iteration_length = self.params.get('iteration_length', default=1024)
+        runtime = self.params.get('runtime', default=5)
+        verbose = self.params.get('verbose', default=0)
+        precompute_random = self.params.get('precompute_random', default=0)
+        intermediate_stats = self.params.get('intermediate_stats', default=0)
+
+        if not (self.params.get('cache_size')):
+            args = '-p %s -c %s -r %s -l %s -t %s' % (pcpu, ccpu, random_seed,
+                                                      iteration_length, runtime)
+        else:
+            cache_size = self.params.get('cache_size', default=32768)
+            args = '-p %s -c %s -r %s -s %s -t %s' % (pcpu, ccpu, random_seed,
+                                                      cache_size, runtime)
+        if (verbose == 1):
+            args += ' --verbose'
+        if (precompute_random == 1):
+            args += ' --precompute-random'
+        if (intermediate_stats == 1):
+            args += ' --intermediate-stats'
+
+        process.system_output('%s %s/producer_consumer %s'
```

Comment:
Please handle this command for failure The purpose of a test must be to fail atleast once in the script.